### PR TITLE
Change to CONTRIBUTING file and docs callouts

### PR DESCRIPTION
**Pull request by theletterf**
- Adding the docs paragraph to CONTRIBUTING.md, as per GDI Specs PR https://github.com/signalfx/gdi-specification/pull/129
- Adding callouts that point to equivalent official docs (when applicable)
- Adding README file in docs folder 

These changes were discussed and approved by @flands.

### CODE REVIEW

File: docs/webengine-attributes.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-> The official Splunk documentation for this page is [Webengine attributes](https://docs.splunk.com/Observability/gdi/get-data-in/application/java/configuration/java-otel-metrics-attributes.html#middleware-attributes-java-otel).
+> The official Splunk documentation for this page is [Middleware attributes](https://docs.splunk.com/Observability/gdi/get-data-in/application/java/configuration/java-otel-metrics-attributes.html#middleware-attributes-java-otel). For instructions on how to contribute to the docs, see [CONTRIBUTE.md](../CONTRIBUTE.md).
```
Comment (iNikem) [inline docs/webengine-attributes.md:1]: Agent does not produce `middleware` attributes anymore, it produces `webengine`

Comment (theletterf) [reply docs/webengine-attributes.md]: Fixed!

**Review by mateuszrzeszutek — APPROVED** [review]

**Review by iNikem — APPROVED** [review]

File: docs/advanced-config.md [hunk 2]
```diff
@@ -1,4 +1,4 @@
-> The official Splunk documentation for this page is [Configure the Java agent](https://docs.splunk.com/Observability/gdi/get-data-in/application/java/configuration/advanced-java-otel-configuration.html).
+> The official Splunk documentation for this page is [Configure the Java agent](https://docs.splunk.com/Observability/gdi/get-data-in/application/java/configuration/advanced-java-otel-configuration.html). For instructions on how to contribute to the docs, see [CONTRIBUTE.md](../CONTRIBUTE.md).
```
Comment (jkwatson) [inline docs/advanced-config.md:1]:
```suggestion
> The official Splunk documentation for this page is [Configure the Java agent](https://docs.splunk.com/Observability/gdi/get-data-in/application/java/configuration/advanced-java-otel-configuration.html). For instructions on how to contribute to the docs, see [CONTRIBUTING.md](../CONTRIBUTING.md).
```

File: docs/correlating-traces-with-logs.md [hunk 3]
```diff
@@ -1,4 +1,4 @@
-> The official Splunk documentation for this page is [Connect Java application trace data with logs](https://docs.splunk.com/Observability/gdi/get-data-in/application/java/instrumentation/connect-traces-logs.html#nav-Connect-trace-data-with-logs).
+> The official Splunk documentation for this page is [Connect Java application trace data with logs](https://docs.splunk.com/Observability/gdi/get-data-in/application/java/instrumentation/connect-traces-logs.html#nav-Connect-trace-data-with-logs). For instructions on how to contribute to the docs, see [CONTRIBUTE.md](../CONTRIBUTE.md).
```
Comment (jkwatson) [inline docs/correlating-traces-with-logs.md:1]:
```suggestion
> The official Splunk documentation for this page is [Connect Java application trace data with logs](https://docs.splunk.com/Observability/gdi/get-data-in/application/java/instrumentation/connect-traces-logs.html#nav-Connect-trace-data-with-logs). For instructions on how to contribute to the docs, see [CONTRIBUTING.md](../CONTRIBUTING.md).
```

File: docs/advanced-config.md (hunk 2 above)
Comment (jkwatson) [reply docs/advanced-config.md:1]: also note the mis-spelled name in all the other places. :)

Comment (theletterf) [reply docs/advanced-config.md]: Well spotted! Fixed in all instances.

**Review by mateuszrzeszutek — APPROVED** [review]

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -30,6 +30,17 @@ When filing an issue, please do *NOT* include:
 - Internal identifiers such as JIRA tickets
 - Any sensitive information related to your environment, users, etc.
 
+## Documentation
+
+The Splunk Observability documentation is hosted on https://docs.splunk.com/Observability,
```
Comment (jkwatson) [inline CONTRIBUTING.md:35]:
```suggestion
The Splunk Observability documentation is hosted at https://docs.splunk.com/Observability,
```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -30,6 +30,17 @@ When filing an issue, please do *NOT* include:
...
+which contains all the prescriptive guidance for Splunk Observability products. 
+Prescriptive guidance consists of step-by-step instructions, conceptual material,
+and decision support for customers. Reference documentation and development 
+documentation is hosted on this repository.
+
+You can send feedback about Splunk Observability docs by clicking the Feedback 
+button on any of our documentation pages.
```
Comment (jkwatson) [inline CONTRIBUTING.md:42]:
```suggestion
button on any of our documentation pages on the official docs site.
```

File: CONTRIBUTING.md (hunk 4 above)
Comment (theletterf) [reply CONTRIBUTING.md]: Quite the debate, interestingly: https://english.stackexchange.com/questions/8226/on-website-or-at-website

**Comment by theletterf** [comment]
@jkwatson Not sure why some tests failed.

File: CONTRIBUTING.md (hunk 4 above)
Comment (jkwatson) [reply CONTRIBUTING.md]: lol

**Comment by jkwatson** [comment]
> @jkwatson Not sure why some tests failed.

/shrug No idea!

**Review by jkwatson — APPROVED** [review]

File: docs/webengine-attributes.md [hunk 6]
```diff
@@ -1,5 +1,5 @@
-> The official Splunk documentation for this page is [Webengine attributes](https://docs.splunk.com/Observability/gdi/get-data-in/application/java/configuration/java-otel-metrics-attributes.html#middleware-attributes-java-otel).
-
+> The official Splunk documentation for this page is [WebEngine attributes](https://docs.splunk.com/Observability/gdi/get-data-in/application/java/configuration/java-otel-metrics-attributes.html#middleware-attributes-java-otel). For instructions on how to contribute to the docs, see [CONTRIBUTING.md](../CONTRIBUTING#documentation.md).
+git
```
Comment (laurit) [inline docs/webengine-attributes.md:2]: this looks like an accidental change

**Pull request merged**